ui_framework/components/modal/confirm_modal.test.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import sinon from 'sinon';
+import { mount } from 'enzyme';
+
+import { requiredProps } from '../../test/required_props';
+
+import {
+  KuiConfirmModal,
+} from './confirm_modal';
+
+const onConfirm = sinon.spy();
+const onCancel = sinon.spy();
+
+test('renders KuiConfirmModal', () => {
+  const component = mount(<KuiConfirmModal
+    message="This is a confirmation modal example"
+    title="A confirmation modal"
+    onCancel={() => {}}
+    onConfirm={onConfirm}
+    cancelButtonText="Cancel Button Text"
+    confirmButtonText="Confirm Button Text"
+    { ...requiredProps }
+  />);
+  expect(component).toMatchSnapshot();
+
+  component.find('[data-test-subj="confirmModalConfirmButton"]').simulate('click');
+  sinon.assert.calledOnce(onConfirm);
+});
+
+test('onConfirm', () => {
+  const component = mount(<KuiConfirmModal
+    message="This is a confirmation modal example"
+    title="A confirmation modal"
+    onCancel={onCancel}
+    onConfirm={onConfirm}
+    cancelButtonText="Cancel"
+    confirmButtonText="Confirm"
+    { ...requiredProps }
+  />);
+  onConfirm.reset();
+  onCancel.reset();
```

Comment:
After looking through https://facebook.github.io/jest/docs/setup-teardown.html, I think we should take advantage of the `beforeEach` setup function.

Can we change the spies to be `let` variables, and just reassign them inside of a `beforeEach`?

```javascript
let onConfirm, onCancel;

beforeEach(() => {
  onConfirm = sinon.spy();
  onCancel = sinon.spy();
});
```